src/store/reducers/posting-project-user/project/project.js

```diff
@@ -1,18 +1,79 @@
 import {
-    SET_PROJECT
-} from '../../../actions/posting-project-user/project/project'
+  SET_PROJECT,
+  ADD_NAME,
+  ADD_DESCRIPTION,
+  ADD_LOCATION,
+  ADD_DEADLINE,
+  ADD_QUESTION,
+  ADD_BENEFIT,
+  ADD_REQUIREMENT,
+  ADD_AVATAR,
+} from "../../../actions/posting-project-user/project/project";
 
 const initialState = {
-    projects: [],
+  projects: [],
+  name: null,
+  description: null,
+  location: null,
+  deadline: null,
+  question: null,
+  benefit: null,
+  requirement: null,
+  avatar: null,
 };
 
 export const projectReducer = (state = initialState, action) => {
-    switch(action.type){
-        case SET_PROJECT:
-            return{
-                ...state,
-                projects: action.projectData
-            }
-        default: return state;
-    }
-}
\ No newline at end of file
+  switch (action.type) {
+    case SET_PROJECT:
+      return {
+        ...state,
+        projects: action.projectData,
+      };
+    case ADD_NAME:
+      console.log("Name dispatched.");
+      return {
+        ...state,
+        name: action.payload,
+      };
+    case ADD_DESCRIPTION:
+      console.log("Description dispatched.");
+      return {
+        ...state,
+        description: action.payload,
+      };
+    case ADD_LOCATION:
+      console.log("Location dispatched.");
+      return {
+        ...state,
+        location: action.payload,
+      };
+    case ADD_DEADLINE:
+      console.log("Deadline dispatched.");
+      return {
+        ...state,
+        deadline: action.payload,
+      };
+    case ADD_QUESTION:
+      return {
+        ...state,
+        question: action.payload,
+      };
+    case ADD_BENEFIT:
+      return {
+        ...state,
+        benefit: action.payload,
+      };
+    case ADD_REQUIREMENT:
+      return {
+        ...state,
+        requirement: action.payload,
+      };
+    case ADD_AVATAR:
+      return {
+        ...state,
+        avatar: action.payload,
+      };
```

Comment:
**Redundant code**
- Doing this will create redundant and separated piece of data. Meanwhile, we want those data to be grouped into an object _**Please check the DFD**_
- Remember that, when we fetch the data, it will contain all the properties, we should group them into 1 object (aka hint: **class**) and just send them to the store.
- Also, I don't know this code works or not (It seems don't). I don't see the **payload** nowhere